packages/hardhat-core/src/internal/hardhat-network/provider/modules/eth.ts

```diff
@@ -1318,47 +1349,138 @@ export class EthModule {
-
-    await this._logTransactionTrace(tx, trace, block, blockResult);
+    let multipleTransactions = false;
+    let sentTxResult: RunTransactionResult | undefined;
+    let sentTxIndex: number | undefined;
+    let error: Error | undefined;
 
-    if (trace !== undefined) {
-      await this._runHardhatNetworkMessageTraceHooks(trace, false);
+    if (results.length > 1) {
+      this._logMultipleBlocksWarning();
     }
 
-    this._logConsoleLogMessages(consoleLogMessages);
+    for (const result of results) {
+      const { block, blockResult, traces } = result;
+      let additionalIndent = false;
+
+      if (results.length === 1 && block.transactions.length > 1) {
+        this._logMultipleTransactionsWarning();
+        this._logBlockNumber(block);
+        multipleTransactions = true;
+        additionalIndent = true;
+      } else if (results.length > 1) {
+        this._logBlockNumber(block);
+        multipleTransactions = true;
+        additionalIndent = true;
+      }
 
-    if (error !== undefined) {
-      if (this._throwOnTransactionFailures) {
-        throw error;
+      for (let i = 0; i < block.transactions.length; i++) {
+        const blockTx = block.transactions[i];
+        const trace = traces[i];
+        let isSentTx = false;
+
+        if (bufferToRpcData(blockTx.hash(true)) === txHash) {
+          sentTxResult = result;
+          sentTxIndex = i;
+          error = trace.error;
+
+          if (results.length > 1 || block.transactions.length > 1) {
+            isSentTx = true;
+          }
+        }
+
+        await this._logTransactionTrace(
+          blockTx,
+          trace.trace,
+          block,
+          blockResult,
+          additionalIndent,
+          isSentTx
+        );
+
+        // Prints an empty line between transactions listed in a block
+        if (i < block.transactions.length && block.transactions.length > 1) {
+          this._logger.logInfo("");
+        }
+
+        if (trace.trace !== undefined) {
+          await this._runHardhatNetworkMessageTraceHooks(trace.trace, false);
+        }
+
+        this._logConsoleLogMessages(trace.consoleLogMessages);
+
+        if (trace.error !== undefined) {
+          this._logError(trace.error);
+        }
       }
+    }
 
-      // TODO: This is a little duplicated with the provider, it should be
-      //  refactored away
-      // TODO: This will log the error, but the RPC method won't be red
-      this._logError(error);
+    if (
+      multipleTransactions &&
+      sentTxResult !== undefined &&
+      sentTxIndex !== undefined
+    ) {
+      const { block, blockResult, traces } = sentTxResult;
+      const trace = traces[sentTxIndex];
+
+      this._logger.logInfo(chalk.bold.yellow("\nCurrently sent transaction:"));
+      await this._logTransactionTrace(
+        tx,
+        trace.trace,
+        block,
+        blockResult,
+        false,
+        true
+      );
+    }
+
+    if (error !== undefined) {
+      throw error;
     }
```

Comment:
What happened to the `_throwOnTransactionFailures` check? 
```suggestion
    if (error !== undefined && this._throwOnTransactionFailures) {
      throw error;
    }
```
The suggestion above probably does not solve the problem completely. We should only throw an error if it concerns the **sent transaction**.

Reply:
It was lost somewhere during development 😅 We only throw errors if it concerns the currently sent transaction! 😎 `error` variable is set only when the hash of transaction, which throws it, equals the sent transaction.